### PR TITLE
Create new utterance instance every time we speak

### LINKED ISSUE #4: p5.speech only speaks once in Firefox 49.0.1
When running examples codes or my own code, p5.speech only speaks the first time it is called. All following calls are ignored.

### PR DESCRIPTION
**Pull request by Zalastax**
Firefox will not schedule an utterance twice so we need to create a new utterance every time something is to be spoken.
Closes #4

### CODE REVIEW

**Comment by rev3rend** [comment]
this is a firefox issue, not a web speech issue... so the real thing to do is get mozilla to fix their browser, instead of us fixing our library.

**Comment by meiamsome** [comment]
I believe the inverse is actually true.

From the specification: https://w3c.github.io/speech-api/webspeechapi.html#tts-methods

>  The SpeechSynthesis object takes exclusive ownership of the SpeechSynthesisUtterance object. Passing it as a speak() argument to another SpeechSynthesis object should throw an exception. (For example, two frames may have the same origin and each will contain a SpeechSynthesis object.)

This implies (At least to me) that using the utterance object in any way after passing it to be synthesised is a violation of the API. Note that despite it specifying it should throw an exception when passed to **another** SpeechSynthesis object it also says that it takes exclusive ownership, and that editing of the text property after this point is undefined behaviour.

**Comment by Zalastax** [comment]
I'd re-open the PR if I could, but the maintainers have to do that. Closing the PR that quickly feels a bit rude.
Even if it was a problem with Mozilla this is not the avenue for pushing that agenda. A library user just cares about it working and that should be the priority.

**Comment by rev3rend** [comment]
i'm gonna ping google for clarification on this.  I think the way it's written in the API is confusing and is missing a step in explaining what happens.  My understanding when i worked with them on this was that a SpeechSynthesisUtterance gets auto-created under the hood when you tell a SpeechSythesis object to speak() a vanilla string, instead of an explicitly created SpeechSynthesisUtterance.  That's what I mean when I say I think this is a Firefox bug.

Making an SpeechSynthesisUtterance explicitly inside the library doesn't make sense to me in this context, because the SpeechSynthesisUtterance is really intended to help you with caching commonly used phrases (they contain all the metadata (rate settings, phonemes, etc.) that the synthesizer needs that stop just short of the actual synthesis, so it's more efficient if your software has only like ten phrases it's gonna use over and over.  I could see us adding some groovy functionality where you could pass an Array or JSON or whatever that the object would then bust out into a bunch of cached utterances, but having a new SpeechSynthesisUtterance for every phrase is what the SpeechSynthesis object is supposed to do by default, so if it's not working, then something else seems wrong to me.

**Comment by rev3rend** [comment]
wasn't meaning to be rude, btw. i wrote this thing over two years ago, and i have literally no idea who any of you are.  this library is used in several dozen accessibility projects for years and this has never been reported.  i'm happy to entertain putting in an explicit utterance for every speech line to fix firefox, but i need to make sure it doesn't break anything in chrome, safari, etc. first.  hang in there and i'll take a look.

File: lib/p5.speech.js [hunk 1]
```diff
@@ -166,38 +136,66 @@
...
+    this.synth.speak(utterance);
+  };
+
+  var speechToUtteranceProperties = {
+    interrupt: "interrupt",
+    onLoad: "onload",
+    onStart: "onstart",
+    onPause: "onpause",
+    onResume: "onresume",
+    onEnd: "onend",
```
Comment (meiamsome) [inline lib/p5.speech.js]: I don't think these events need to be passed down - they bubble up to the SpeechSynthesiser instance automatically?

**Comment by rev3rend** [comment]
ah i see what you mean!  so are things like rate/pitch sticky if you recreate the utterance?

let me take a crack at this and see if i can figure it out.

**Pull request closed without merging**